x-pack/test/plugin_functional/test_suites/global_search/global_search_api.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+import { NavigableGlobalSearchResult } from '../../../../plugins/global_search/public/services/types';
+// import { GlobalSearchTestApi } from '../../plugins/global_search_test/public/types';
+
+export default function({ getPageObjects, getService }: FtrProviderContext) {
+  const pageObjects = getPageObjects(['common']);
+  const browser = getService('browser');
+
+  const findResultsWithAPI = async (t: string): Promise<NavigableGlobalSearchResult[]> => {
+    return browser.executeAsync(async (term: string, cb: Function) => {
+      window.__globalSearchTestApi.findAll(term).then(results => {
+        cb(results);
+      });
+    }, t) as any; // executeAsync signature is broken. return type should be inferred from the cb param.
```

Comment:
Creating a 'full' test app displaying results on a page and then asserting on the page content felt like too much work for little gain compared to direct API call invocation. I can still do it if we feel like this is necessary.